### PR TITLE
Bridge: Support Wormhole

### PR DESCRIPTION
**Pull request by lgalende**

### CODE REVIEW

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    uint256 private constant ENCODED_DATA_LENGTH = 0;
+
+    // List of Wormhole chain IDs
+    uint16 private constant ETHEREUM_DOMAIN = 2;
+    uint16 private constant POLYGON_DOMAIN = 5;
+    uint16 private constant ARBITRUM_DOMAIN = 23;
+    uint16 private constant OPTIMISM_DOMAIN = 24;
+    uint16 private constant BSC_DOMAIN = 4;
+    uint16 private constant FANTOM_DOMAIN = 10;
+    uint16 private constant AVALANCHE_DOMAIN = 6;
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:38]: I think we can call these `X_WORMHOLE_NETWORK_ID` or similar, `DOMAIN` was a keyword used by Axelar, but not standard at all

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+import '@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol';
+
+import '@mimic-fi/v2-helpers/contracts/utils/ERC20Helpers.sol';
+
+import '../interfaces/IWormhole.sol';
+
+/**
+ * @title WormholeConnector
+ * @dev Interfaces with Wormhole to bridge tokens
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:25]:
```suggestion
 * @dev Interfaces with Wormhole to bridge tokens through CCTP
```

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    function _bridgeWormhole(
+        uint256 chainId,
+        address token,
+        uint256 amountIn,
+        uint256, /* minAmountOut */
+        address recipient,
+        bytes memory data
+    ) internal {
+        require(data.length == ENCODED_DATA_LENGTH, 'WORMHOLE_INVALID_DATA_LENGTH');
+        uint16 domain = _getWormholeChainDomain(chainId);
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:77]:
```suggestion
        uint16 wormholeNetworkId = _getWormholeChainDomain(chainId);
```

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            bytes32(uint256(uint160(recipient))) // convert from address to bytes32
+        );
+    }
+
+    /**
+     * @dev Internal function to tell the chain domain based on a chain ID
+     * @param chainId ID of the chain being queried
+     * @return Chain domain associated to the requested chain ID
+     */
+    function _getWormholeChainDomain(uint256 chainId) private pure returns (uint16) {
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:94]: What about `_getWormholeNetworkId`, I think they call it that way actually.

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            token,
+            amountIn,
+            0, // don't swap to native token
+            domain,
+            bytes32(uint256(uint160(recipient))) // convert from address to bytes32
+        );
+    }
+
+    /**
+     * @dev Internal function to tell the chain domain based on a chain ID
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:90]:
```suggestion
     * @dev Internal function to tell the Wormhole network ID based on a chain ID
```

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+     * @param chainId ID of the chain being queried
+     * @return Chain domain associated to the requested chain ID
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:92]:
```suggestion
     * @return Wormhole network ID associated to the requested chain ID
```

**Review by facuspagnuolo — APPROVED** [review]
Congrats! Super clean and straightforward! Nice first contribution 👏 
Left a few comments around naming and inline docs only

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 7]
```diff
@@ -62,19 +62,23 @@ contract WormholeConnector {
...
         uint256 amountIn,
-        uint256, /* minAmountOut */
+        uint256 minAmountOut,
         address recipient,
         bytes memory data
     ) internal {
         require(data.length == ENCODED_DATA_LENGTH, 'WORMHOLE_INVALID_DATA_LENGTH');
         uint16 wormholeNetworkId = _getWormholeNetworkId(chainId);
+        uint256 relayerFee = IWormhole(wormholeCircleRelayer).relayerFee(wormholeNetworkId, token);
+        // solhint-disable-next-line reason-string
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:80]: This is due to a reason, let's try to shorten it up-to 32 bytes

**Review by facuspagnuolo — APPROVED** [review]
👏

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+/**
+ * @title WormholeConnector
+ * @dev Interfaces with Wormhole to bridge tokens through CCTP
+ */
+contract WormholeConnector {
+    // Expected data length when bridging with Wormhole: none
+    uint256 private constant ENCODED_DATA_LENGTH = 0;
+
+    // List of Wormhole chain IDs
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:31]:
```suggestion
    // List of Wormhole network IDs
```

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+        uint256 amountIn,
+        uint256 minAmountOut,
+        address recipient,
+        bytes memory data
+    ) internal {
+        require(data.length == ENCODED_DATA_LENGTH, 'WORMHOLE_INVALID_DATA_LENGTH');
+        uint16 wormholeNetworkId = _getWormholeNetworkId(chainId);
+        uint256 relayerFee = IWormhole(wormholeCircleRelayer).relayerFee(wormholeNetworkId, token);
+        // solhint-disable-next-line reason-string
+        require(minAmountOut <= amountIn - relayerFee, 'WORMHOLE_MIN_AMOUNT_GT_AMOUNT_IN_MINUS_RELAYER_FEE');
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/connectors/WormholeConnector.sol:81]:
```suggestion
        require(minAmountOut <= amountIn - relayerFee, 'WORMHOLE_MIN_AMOUNT_OUT_TOO_BIG');
```

File: packages/bridge-connector/contracts/interfaces/IWormhole.sol [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+interface IWormhole {
+    function transferTokensWithRelay(
+        address token,
+        uint256 amount,
+        uint256 toNativeTokenAmount,
+        uint16 targetChain,
+        bytes32 targetRecipientWallet
+    ) external payable returns (uint64 messageSequence);
+
+    function relayerFee(uint16 chainId_, address token) external view returns (uint256);
```
Comment (facuspagnuolo) [inline packages/bridge-connector/contracts/interfaces/IWormhole.sol:26]:
```suggestion
    function relayerFee(uint16 chainId, address token) external view returns (uint256);
```

File: packages/bridge-connector/contracts/connectors/WormholeConnector.sol (hunk 7 above)
Comment (lgalende) [reply packages/bridge-connector/contracts/connectors/WormholeConnector.sol]: Done

**Pull request merged**